client/my-sites/sites/sites.jsx

```diff
@@ -93,4 +108,13 @@ export default React.createClass( {
 			</Main>
 		);
 	}
-} );
+}
+
+export default connect(
+	( state ) => ( {
```

Comment:
What do you think of transforming `SitesList` into an abstraction layer over `state.sites`. So we would keep the poller for now which would update `state.sites` instead of `SitesList` and any call to existing methods in `SitesList` would be a wrapper to a selector + some custom logic. For instance:
```
import { getSite } from 'state/sites/selectors';

SitesList.prototype.getSite = function( siteID ) {
	if ( ! siteID ) {
		return false;
	}

	return new Site( getSite( siteID ) );
};
```

`SitesList#getSite` is a bit more complex than that actually since you can select a site using the slug, the domain or the wpcom url.